### PR TITLE
120036 cfg custom unmarshaling

### PR DESCRIPTION
**Pull request by LinMAD**
Add support to `DatadogConfig` to work with Golang vanilla JSON tags.
Issue related to bool type in DatadogConfig that will return errors on parsing after calling `json.Unmarshal`

### CODE REVIEW

File: boostrap_test.go [hunk 1]
```diff
@@ -8,12 +8,12 @@ import (
 )
 
 func TestDatadog(t *testing.T) {
-	ddCfg := config.DatadogConfig{}
+	ddCfg := new(config.DatadogConfig)
 
 	err := StartDatadog(ddCfg, ConnectionTypeHTTP)
```
Comment (LinMAD) [inline boostrap_test.go:13]: FYI, this test was broken not sure why in CI it's not executed anymore but `StartDatadog accepts the interface now, and then the structure must be passed by a pointer.

**Review by nhhagen — CHANGES_REQUESTED** [review]
I think we need to look closer at this. Like @bendiknesbo is stating this will break the code everywhere that `DatadogConfig` is used as a value.

Are we sure that we want to commit our selves to support JSON as a way of configuring `go-datadog-lib`. Our new helm chart for deploying services uses environment variables to configure Datadog. I think we should move away from supporting custom configuration methods for `go-Datadog-lib`.

We should probably not discuss the future of `go-datadog-lib` in this PR.

**Comment by nhhagen** [comment]
I removed the major version label so that we do not accidentally bump the major version of go-datadog-lib.

@arunpoudel how do I disable the major version label. Just breaking the API is not enough to bump a major version in Go. We also need to update the module name as well.

File: boostrap_test.go (hunk 1 above)
Comment (LinMAD) [reply boostrap_test.go:13]: Could be was not thinking about that actually

**Comment by LinMAD** [comment]
@nhhagen Closing this PR since in the future there are plans to provide the functionality to load configuration with env vars for Datadog so at this point implementing unmarshal does not give much value, since it will be still open for modification via the interface.

**Comment by arunpoudel** [comment]
> I removed the major version label so that we do not accidentally bump the major version of go-datadog-lib.
> 
> @arunpoudel how do I disable the major version label. Just breaking the API is not enough to bump a major version in Go. We also need to update the module name as well.

I will work on it next techdebt day.

https://github.com/coopnorge/github-workflow-release-drafter/issues/45

**Pull request closed without merging**